src/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.cs

```diff
@@ -1117,6 +1117,35 @@ public AssemblyReference GetAssemblyReference(AssemblyReferenceHandle handle)
             return new AssemblyReference(this, handle.Value);
         }
 
+        public AssemblyName GetAssemblyName()
+        {
+            StringHandle nameHandle = AssemblyTable.GetName();
+            string name = (!nameHandle.IsNil) ? GetString(nameHandle) : null;
+            Version version = AssemblyTable.GetVersion();
+
+            StringHandle cultureHandle = AssemblyTable.GetCulture();
+            string cultureName = (!cultureHandle.IsNil) ? GetString(cultureHandle) : null;
+
+            AssemblyFlags flags = AssemblyTable.GetFlags();
+
+            bool hasPublicKey = (flags & AssemblyFlags.PublicKey) != 0;
+            BlobHandle publicKeyHandle = AssemblyTable.GetPublicKey();
+            byte[] publicKeyOrToken = !publicKeyHandle.IsNil
+                ? GetBlobBytes(publicKeyHandle)
+                : null;
+            
+            var assemblyName = new AssemblyName(name)
+            {
+                Version = version,
+                CultureName = cultureName
+            };
+
+            assemblyName.SetPublicKey(publicKeyOrToken);
+            //assemblyName.Flags = AssemblyTable.GetFlags();
+            //assemblyName.HashAlgorithm = AssemblyTable.GetHashAlgorithm();
```

Comment:
thanks @safern could you please now take a look at the new changeset and let me know what you think?